lib/url-helpers.js

```diff
@@ -166,3 +175,11 @@ export const isTrustedRedirectHost = host => {
     return host === domain || host.endsWith(`.${domain}`);
   });
 };
+
+export const addParentToURLIfMissing = (router, account, url) => {
+  const query = router.query;
+  const urlWithParent = getCollectivePageRoute(account) + url;
+  if ([CollectiveType.EVENT, CollectiveType.PROJECT].includes(account?.type) && !query.parentCollectiveSlug) {
+    router.push(urlWithParent, undefined, { shallow: true });
```

Comment:
:question: Are we preserving the query parameters here? When I go to:
> https://staging.opencollective.com/big-party-2-16658ev/expenses?type=INVOICE&status=PENDING

I expect to be redirected to:
> https://staging.opencollective.com/captainfact_io/events/big-party-2-16658ev/expenses?type=INVOICE&status=PENDING

Reply:
Thanks. I've changed the code to add support for this as well. 👍🏽